### PR TITLE
Add ES|QL signum function

### PR DESCRIPTION
**Pull request by ioanatia**
https://github.com/elastic/elasticsearch/issues/98545

Implements the signum function.

```
PUT testidx
{
  "mappings": {
    "properties": {
      "float_field": {
        "type": "float"
      }
    }
  }
}

POST testidx/_doc
{ "float_field": 25 }

POST testidx/_doc
{ "float_field": -100.0 }

POST testidx/_doc
{ "float_field": 0 }

POST testidx/_doc
{ "float_field": null }


POST _query
{
  "query": """
  FROM testidx|
  EVAL s = signum(float_field)
  """
}
```

result:
```
{
  "columns": [
    {
      "name": "float_field",
      "type": "double"
    },
    {
      "name": "s",
      "type": "double"
    }
  ],
  "values": [
    [
      25,
      1
    ],
    [
      -100,
      -1
    ],
    [
      0,
      0
    ],
    [
      null,
      null
    ]
  ]
}
```

### CODE REVIEW

File: x-pack/plugin/esql/qa/testFixtures/src/main/resources/math.csv-spec [hunk 1]
```diff
@@ -1249,6 +1249,96 @@ i:ul                | c:ul                | f:ul
...
+;
+
+signumOfNegativeInteger#[skip:-8.13.99,reason:new scalar function added in 8.14]
+row i = -100 | eval s = signum(i);
+
+i:integer | s:double
+-100       | -1.0
+;
+
+signumOfZeroInteger#[skip:-8.13.99,reason:new scalar function added in 8.14]
```
Comment (nik9000) [inline x-pack/plugin/esql/qa/testFixtures/src/main/resources/math.csv-spec:1279]: I've been leaning towards putting these in `floats.csv-spec` and `ints.csv-spec` and `unsigned_long.csv-spec`.

File: docs/reference/esql/functions/layout/signum.asciidoc [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// This is generated by ESQL's AbstractFunctionTestCase. Do no edit it. See ../README.md for how to regenerate it.
```
Comment (nik9000) [inline docs/reference/esql/functions/layout/signum.asciidoc:1]: If you link to this file in `math-functions.asciidoc` then we'll render the generated docs.

Comment (nik9000) [reply docs/reference/esql/functions/layout/signum.asciidoc:1]: Would you mind updating the `package-info.java` file with instructions to do this? I should have done it when I wrote the code generation, but I forgot.

**Review by nik9000 — APPROVED** [review]

File: docs/reference/esql/functions/layout/signum.asciidoc (hunk 2 above)
Comment (ioanatia) [reply docs/reference/esql/functions/layout/signum.asciidoc:1]: Done and done! the `package-info.java` instructions are great btw!

**Review by astefan — DISMISSED** [review]
Thank you for your contribution.

Having csv-spec tests that only use `row` is not enough.
Take a look at other csv-spec files where `from` command is used and try to have more complex queries with nested functions, using the `signum` function in all other commands (sort, eval, filter, stats etc).

**Comment by nik9000** [comment]
> Having csv-spec tests that only use `row` is not enough.

Ah! I should have caught that one.

**Review by nik9000 — APPROVED** [review]

**Comment by ioanatia** [comment]
@astefan - more tests have been added and Nik reviewed the PR again yesterday.
I merged it since I wanted to avoid any new merge conflicts, but if there is any additional feedback from you let me know.

**Pull request merged**